judge/tasks/submission.py

```diff
@@ -18,11 +18,40 @@ def apply_submission_filter(queryset, id_range, languages, results):
         queryset = queryset.filter(language_id__in=languages)
     if results:
         queryset = queryset.filter(result__in=results)
-    queryset = queryset.exclude(locked_after__lt=timezone.now()) \
+    queryset = queryset.exclude(is_archived=True) \
+                       .exclude(locked_after__lt=timezone.now()) \
                        .exclude(status__in=Submission.IN_PROGRESS_GRADING_STATUS)
     return queryset
 
 
+@shared_task(bind=True)
+def archive_locked_submissions(self, problem_id):
+    submissions = \
+        Submission.objects.filter(problem_id=problem_id, is_archived=False, locked_after__lt=timezone.now())
+
+    with Progress(self, submissions.count(), stage=_('Modifying submissions')) as p:
+        archived = 0
+        for submission in submissions.iterator():
+            submission.is_archived = True
+            submission.save(update_fields=['is_archived'])
+            archived += 1
+            if archived % 10 == 0:
+                p.done = archived
```

Comment:
Does this loop result in N database calls? Can't we do this with one database call?

Reply:
I don't think this results in DB calls, rather it's just used for the progress bar (someone please check me on that).

Reply:
i copy pasted from `rescore_problem`. i guess `rescore_problem` is poorly designed too.

Reply:
this code no longer exists
i went with a different approach (see new toggle in `templates/problem/manage_submission.html`)